blockprod/src/detail/block_maker.rs

```diff
@@ -147,10 +198,11 @@ impl BlockMaker {
     /// 2. A new tip is now on chainstate, indicating that there's no point in continuing to mine/stake at that tip
     pub async fn run(&mut self) -> Result<(), BlockProductionError> {
         let accumulator = self.collect_transactions().await?;
+        let transactions = accumulator.transactions();
 
         // TODO: do we want to introduce a separate executor for this loop to avoid starving other tasks?
         loop {
-            let block = self.make_block(self.current_tip_id, &*accumulator)?;
+            let block = self.make_block(self.current_tip_id, transactions).await?;
```

Comment:
Keep in mind that the way it's done now, mining will *never* exit unless mining is finished, which is practically impossible. So cancelling won't work.

Notice that the cancel function works assuming that the make_block function doesn't really take a long time to finish.

Reply:
Ah, yes this was why I got confused about u128::MAX. You're right... I'll have to think about a nice way to solve this.